### ISSUE TITLE
Attack Components

### ISSUE THREAD

**Issue by stormhalter** [jkachhad/Stormhalter#194]
Implement additional attack mechanics such as Disarms, Silences, Life-steal, Mana-Drain, Visibility Reduction, Perception Distance, Confusion (Map rotations) [Suggested by Lux]

---
[View on Discord](https://discord.com/channels/1056432310632841276/1530626698930819083/1530626698930819083)

---
[View on Discord](https://discord.com/channels/577503035526348810/1532164832151736462/1532164832151736462)

**Comment by SlysterLOK** [comment]
Disarms - Meh. Possibly Lair Only. Otherwise in hunting areas that is instant death later on since your weapon is your main defense.

Silence - Again Possibly Lair Only. Either way a MU will just wait in place until the silence ends.

Life Steal - Yes.

Mana Drain - Depends on how plentiful MRPs are I suppose. Possibly Lair Only.

Confusion - Yes!

Visibility Reduction - That would be great in a new underground cavern or something. Maybe need a light source to see at all?

Perception Distance - Not sure what this is..? To see creatures?

**Comment by jkachhad** [comment]
Correct, for example, you're used to seeing most creatures with a perception of 3. Certain creatures have a visibility distance of 0, for example Bees and Beetles. You can't see them unless they are on your tile.

Perception distance might change how far our you can see an enemy. It would be dangerous to have a debuff that reduces your perception to 0, but there are casters that you have to fizzle.

**Comment by Chilly021** [comment]
Is this all back end code? I'd be interested to add some implementation if I could, but wasn't sure if this is just a suggestion for later.
